app/controllers/user.js

```diff
@@ -38,19 +40,70 @@ const emailIsRegistered = email =>
     throw errors.databaseError(e.message);
   });
 
+const createUser = (user, isAdmin = false) =>
+  bcryptService
+    .hashPassword(user.password)
+    .then(hashedPassword => User.create({ ...user, password: hashedPassword, isAdmin }));
+
 exports.signUp = ({ user }, res, next) =>
   emailIsRegistered(user.email)
     .then(isRegistered => {
       if (isRegistered) throw errors.badRequest(errorMsgs.emailIsAlreadyRegistered);
       else
-        return bcryptService.hashPassword(user.password).then(hashedPassword =>
-          User.create({ ...user, password: hashedPassword }).then(newUser => {
-            logger.info(`User ${newUser.lastName}, ${newUser.firstName} created successfuly`);
+        return createUser(user).then(newUser => {
+          logger.info(`User ${newUser.lastName}, ${newUser.firstName} created successfuly`);
+          res
+            .status(201)
+            .send(newUser)
+            .end();
+        });
+    })
+    .catch(next);
+
+const getUserForToken = token => {
+  try {
+    const decoded = jwt.decode(token);
```

Comment:
Did you know that you can deconstruct the object directly? For example, in this case : 
const { email } = jwt.decode(token);